From ff1456c79cb59906d8496ba3eb7cab6b03981ea0 Mon Sep 17 00:00:00 2001
From: Jon Nettleton <jon@solid-run.com>
Date: Thu, 28 Sep 2023 17:16:54 +0200
Subject: [PATCH 45/45] sr-rz: Consolidate env and bootargs handling

This patch consolidates similar bootargs and
environment partitions for all the rzg and rzv
solidrun platforms. We use a custom handler
and choose the partition depending on if we have
booted from usdhc or emmc. If it is usdhc then
we always use partition 0, if it is eMMC we use
the defined ENV partition in the kconfig. I have
moved this to the first boot partition by default
so the second partition is completely available
for the customers use.

Signed-off-by: Jon Nettleton <jon@solid-run.com>
---
 board/solidrun/rzg-common/rzg-common.c | 11 ++++++++++-
 configs/rzg2l-solidrun_defconfig       |  5 +++--
 configs/rzg2lc-solidrun_defconfig      |  4 ++--
 configs/rzv2l-solidrun_defconfig       |  4 +++-
 4 files changed, 18 insertions(+), 6 deletions(-)

diff --git a/board/solidrun/rzg-common/rzg-common.c b/board/solidrun/rzg-common/rzg-common.c
index e0177808..39c54b93 100644
--- a/board/solidrun/rzg-common/rzg-common.c
+++ b/board/solidrun/rzg-common/rzg-common.c
@@ -130,6 +130,15 @@ void rzg_sd_emmc_init(void)
 	set_bootsource_env(value);
 }
 
+uint mmc_get_env_part(struct mmc *mmc)
+{
+	int value = board_check_sd_emmc();
+	if (value == 1)
+		return 0;
+	else
+		return CONFIG_SYS_MMC_ENV_PART;
+}
+
 #if defined(CONFIG_OF_LIBFDT) && defined(CONFIG_OF_BOARD_SETUP) && defined(CONFIG_OF_SYSTEM_SETUP)
 /*
  * Configure the correct sdhi0 node (eMMC/SD) in device-tree:
@@ -222,5 +231,5 @@ int rzg_preboot_sd_emmc_setup(void *blob, struct bd_info *bd)
 
 	return 0;
 }
+#endif
 
-#endif
\ No newline at end of file
diff --git a/configs/rzg2l-solidrun_defconfig b/configs/rzg2l-solidrun_defconfig
index ab5381f7..34af08df 100644
--- a/configs/rzg2l-solidrun_defconfig
+++ b/configs/rzg2l-solidrun_defconfig
@@ -16,7 +16,7 @@ CONFIG_SPL=n
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
 CONFIG_BOARD_LATE_INIT=y
-CONFIG_BOOTARGS="root=/dev/mmcblk0p2 rw rootwait"
+CONFIG_BOOTARGS="rw rootwait root=/dev/mmcblk0p2"
 CONFIG_SUPPORT_RAW_INITRD=y
 CONFIG_DEFAULT_FDT_FILE="rzg2l-hummingboard-ripple.dtb"
 CONFIG_VERSION_VARIABLE=y
@@ -76,7 +76,8 @@ CONFIG_USB_STORAGE=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_SMBIOS_MANUFACTURER=""
 CONFIG_ENV_IS_IN_MMC=y
-CONFIG_CLK_R9A07G044L=y
+CONFIG_SYS_MMC_ENV_DEV=0
+CONFIG_SYS_MMC_ENV_PART=1
 CONFIG_SYS_I2C_RZG2L_RIIC=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_DISTRO_DEFAULTS=y
diff --git a/configs/rzg2lc-solidrun_defconfig b/configs/rzg2lc-solidrun_defconfig
index 6430c038..1ba38014 100644
--- a/configs/rzg2lc-solidrun_defconfig
+++ b/configs/rzg2lc-solidrun_defconfig
@@ -14,7 +14,7 @@ CONFIG_SPL=n
 CONFIG_FIT=y
 CONFIG_BOARD_LATE_INIT=y
 CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="rw rootwait earlycon root=/dev/mmcblk0p1"
+CONFIG_BOOTARGS="rw rootwait root=/dev/mmcblk0p2"
 CONFIG_USE_BOOTCOMMAND=y
 CONFIG_BOOTCOMMAND="run distro_bootcmd"
 CONFIG_SUPPORT_RAW_INITRD=y
@@ -77,7 +77,7 @@ CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_SMBIOS_MANUFACTURER=""
 CONFIG_ENV_IS_IN_MMC=y
 CONFIG_SYS_MMC_ENV_DEV=0
-CONFIG_SYS_MMC_ENV_PART=2
+CONFIG_SYS_MMC_ENV_PART=1
 CONFIG_SYS_I2C_RZG2L_RIIC=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_MISC=y
diff --git a/configs/rzv2l-solidrun_defconfig b/configs/rzv2l-solidrun_defconfig
index 6524d18a..ff6b8fd4 100644
--- a/configs/rzv2l-solidrun_defconfig
+++ b/configs/rzv2l-solidrun_defconfig
@@ -16,7 +16,7 @@ CONFIG_SPL=n
 CONFIG_FIT=y
 CONFIG_USE_BOOTARGS=y
 CONFIG_BOARD_LATE_INIT=y
-CONFIG_BOOTARGS="root=/dev/mmcblk0p2 rw rootwait"
+CONFIG_BOOTARGS="rw rootwait root=/dev/mmcblk0p2"
 CONFIG_SUPPORT_RAW_INITRD=y
 CONFIG_DEFAULT_FDT_FILE="rzv2l-hummingboard-extended.dtb"
 CONFIG_VERSION_VARIABLE=y
@@ -76,6 +76,8 @@ CONFIG_USB_STORAGE=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_SMBIOS_MANUFACTURER=""
 CONFIG_ENV_IS_IN_MMC=y
+CONFIG_SYS_MMC_ENV_DEV=0
+CONFIG_SYS_MMC_ENV_PART=1
 CONFIG_SYS_I2C_RZG2L_RIIC=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_DISTRO_DEFAULTS=y
-- 
2.41.0

